### DISCUSSION TITLE
How to add a service as tenant scope singletone?

### DISCUSSION THREAD

**Discussion by HengzheLi** [OrchardCMS/OrchardCore#7679] [Q&A]
I want to share one service instance in a tenant. But If I add the service to IOC with AddScoped, the service only can be shared in a request. And if I add the service to IOC with AddSingletone, the service is shared cross tenants. 

Is there a way, like AddTenantSingletone, to share a service in a tenant.

**Comment by jtkech** [comment]
In a regular way `services.AddSingleton<>()` when called in a module / theme startup `ConfigureServices()`, as it is executed (if the related module / feature is enabled) when we compose a given tenant context.

Or from the app by using the following helper, here always executed as the application behaves as a module that is always enabled.

    services.AddOrchardCms()
        .ConfigureServices(tenantServices =>
        {
            tenantServices.AddSingleton
        });
